sdk/search/Azure.Search/README.md

```diff
@@ -35,106 +54,269 @@ To create a new Search Service, you can use the [Azure Portal][create_search_ser
+}
+```
+
+#### Use `SearchDocument` with [C#'s `dynamic` keyword](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/types/using-type-dynamic)
+
+This starts out the same, but the use of `dynamic` makes the code a little
+easier to read.
+```C# Snippet:Azure_Search_Tests_Samples_Readme_Dynamic
+SearchResults<SearchDocument> response = client.Search("luxury");
+foreach (SearchResult<SearchDocument> result in response.GetResults())
+{
+    dynamic doc = result.Document;
+    string id = doc.hotelId;
+    string name = doc.hotelName;
+    Console.WriteLine("{id}: {name}");
+}
 ```
 
+#### Use C# types
+
+We can also decorate our own types with [attributes from `System.Text.Json`](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-how-to)
+```C# Snippet:Azure_Search_Tests_Samples_Readme_StaticType
+public class Hotel
+{
+    [JsonPropertyName("hotelId")]
+    public string Id { get; set; }
+
+    [JsonPropertyName("hotelName")]
+    public string Name { get; set; }
+}
+```
+
+And use them in place of `SearchDocument` when querying.
+
+```C# Snippet:Azure_Search_Tests_Samples_Readme_StaticQuery
+SearchResults<Hotel> response = client.Search<Hotel>("luxury");
+foreach (SearchResult<Hotel> result in response.GetResults())
+{
+    Hotel doc = result.Document;
+    Console.WriteLine($"{doc.Id}: {doc.Name}");
+}
+```
+
+If you're working with a search index and know the schema, creating C# types
+is recommended.
+
+#### SearchOptions
+The `SearchOptions` provide powerful control over the behavior of our queries.
+Let's search for the top 5 luxury hotels with a good rating.
+
+```C# Snippet:Azure_Search_Tests_Samples_Readme_Options
+int stars = 4;
+SearchOptions options = new SearchOptions
+{
+    // Filter to only ratings greater than or equal our preference
+    Filter = SearchFilter.Create($"rating ge {stars}"),
+    Size = 5, // Take only 5 results
+    OrderBy = new[] { "rating desc" } // Sort by rating from high to low
+};
+SearchResults<Hotel> response = client.Search<Hotel>("luxury", options);
+// ...
+``` 
+
+### Adding documents to your index
+
+You can Upload, Merge, UploadOrMerge, and Delete multiple documents from an
+index in a single batched request.  There are [a few special rules for merging](https://docs.microsoft.com/en-us/rest/api/searchservice/addupdate-or-delete-documents#document-actions)
+to be aware of.
+
+```C# Snippet:Azure_Search_Tests_Samples_Readme_Index
+IndexDocumentsBatch<Hotel> batch = IndexDocumentsBatch.Create(
+    IndexDocumentsAction.Upload(new Hotel { Id = "783", Name = "Upload Inn" }),
+    IndexDocumentsAction.Merge(new Hotel { Id = "12", Name = "Renovated Ranch" }));
+
+IndexDocumentsOptions options = new IndexDocumentsOptions { ThrowOnAnyError = true };
+client.IndexDocuments(batch, options);
+```
+
+The request will succeed even if any of the individual actions fails and
+return an `IndexDocumentsResult` for inspection.  There's also a `ThrowOnAnyError`
```

Comment:
Regarding `ThrowOnAnyError` -- I'm not sure the explanation makes sense, I guess because this didn't used to be optional. The idea behind throwing `IndexBatchException` on 207 was to force partial failures to be handled; Otherwise there could effectively be data loss if any failed updates are ignored.

Reply:
We're going to have a lot of fun digging into this one in more detail.  We've been trying _really_ hard not to add other types of exceptions so that using Track 2 means there's one thing you've got to worry about catching.  Without a custom exception, we wouldn't be giving people the information to process/recover.  This was debated a lot during Storage batching and we landed on this pattern.

Reply:
The long-term solution is our "smart batching" idea; That makes this problem go away.

You're right that throwing on 207 without having extra info in the exception is an incomplete scenario, which raises the question -- Why have `ThrowOnAnyError` at all if we're never going to have `IndexBatchException`?